packages/dropdown/src/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,212 @@
+import LeafyGreenProvider, {
+  useDarkMode,
+} from '@leafygreen-ui/leafygreen-provider';
+import Popover, { Align, Justify } from '@leafygreen-ui/popover';
+
+import { DescendantContext, useDescendants } from '../DescendantContext';
+import { DropdownContext } from '../DropdownContext';
+import { HighlightContext } from '../HighlightContext';
+import { handleKeyboardNavigation, useMergeRefs } from '../utils';
+
+import {
+  baseMenuStyle,
+  menuListStyle,
+  menuThemeStyles,
+} from './Dropdown.styles';
+import { DropdownProps, HighlightBehavior } from './Dropdown.types';
+
+export const Dropdown = React.forwardRef(
+  (
+    {
+      align = Align.Bottom,
+      adjustOnMutation = false,
+      children,
+      className,
+      darkMode: darkModeProp,
+      highlightBehavior = HighlightBehavior.Focus,
+      justify = Justify.End,
+      maxWidth,
+      open,
+      portalClassName,
+      portalContainer,
+      popoverZIndex,
+      scrollContainer,
+      setOpen,
+      shouldClose = () => true,
+      spacing = 6,
+      triggerRef,
+      usePortal = true,
+      renderedContext = RenderedContext.Menu,
+      ...rest
+    }: DropdownProps,
+    forwardRef,
+  ) => {
+    const { darkMode, theme } = useDarkMode(darkModeProp);
+    const { ref, ...contextProps } = useDescendants();
+    const dropdownRef = useMergeRefs(forwardRef, ref);
+    const [highlightedRef, setHighlightedRef] = useState<HTMLElement | null>(
+      null,
+    );
+    const [firstOpen, setFirstOpen] = useState(false);
+    const previousOpenState = usePrevious(open);
+
+    // Gets list of registered item refs and filters out disabled items
+    // We use this to accurately cycle highlighted state
+    const enabledRefs = contextProps?.list?.current
+      ?.filter(element => {
+        const id = element._internalId;
+        return !contextProps?.map?.current?.[id].disabled;
+      })
+      .map(element => element.element);
+
+    // Keeps currently highlighted item in state
+    // And manually moves focus to appropriate item if `highlightBehavior` is set to 'focus'
+    const setHighlighted = useCallback(
+      (ref: HTMLElement | null) => {
+        setHighlightedRef(ref);
+
+        if (highlightBehavior === HighlightBehavior.Focus && ref) {
+          ref.focus();
+        }
+      },
+      [highlightBehavior],
+    );
+
+    // Check that this is the first time we're opening the Dropdown
+    // Will use this to move focus to the first element but the check below will be true before
+    // `enabledRefs` has been populated
+    useEffect(() => {
+      if (open && previousOpenState === false) {
+        setFirstOpen(true);
+      }
+    }, [open, previousOpenState, setFirstOpen]);
+
+    // If first open and `enabledRefs` contains the appropriate refs
+    // Move focus to first enabled ref
+    useEffect(() => {
+      if (firstOpen && enabledRefs?.[0]) {
+        setHighlighted(enabledRefs[0]);
+        setFirstOpen(false);
```

Comment:
Maybe I'm not understanding the purpose of this variable, but why would we re-set `firstOpen` back to false?

Reply:
We only want to force highlight to the first element when the menu is opened for the first time